### ISSUE TITLE
The structure that we need to handle stream data

### ISSUE THREAD

**Issue by vahedi25411** [CityOfLosAngeles/automated-walk-bike-counter#93]
In real-time, we need a structure that is responsible for making sure that we have received the stream data effectively and make it ready for the next step which is processing.

What do we need for this structure?

**Comment by vahedi25411** [comment]
I think this has to be implemented in a different thread than the main.

**Comment by ian-r-rose** [comment]
Pseudocode we discussed this week:

```python
class Stream(object):
    def get_frame():
        return frame
    
    def is_finished():
        raise NotImplementedError()
        
    def get_time():
        raise NotImplementedError()

        
class VideoStream(Stream):
    def is_finished():
        return False
    
    def get_time():
        return time.clock()
    
class VideoFile(Stream):
    
    def get_time():
        return self.frames/self.fps


class ObjectTracker():
    
    def run():
        while not stream.is_finished():
            self.process_frame()
            counter.count_objects(self)
            
            if stream.get_time() multiple of 15 minutes:
                counter.export_counts()
            
```

**Comment by vahedi25411** [comment]
@ian-r-rose My question is about the get_time() implementation in VideoStream. As far as I could understand you say that we should get the time form the streaming video, is that right?

**Comment by ian-r-rose** [comment]
No, I was thinking that for streaming video we would use the system clock

**Comment by vahedi25411** [comment]
So, suppose that we have two implementations for get_time() method in VideoFile and VideoStream classes. In our implementation for VideoStream we need a procedure to indicate the time we should export the data. For example, if the interval for exporting the counters is 15 minutes, we should have something like the following pseudo-code:

set the start_time
while True:
    if start_time - current_time >= 15 minutes:
         export the 15 minutes data to the excel file
         start_time = current_time

Do you agree?

**Comment by vahedi25411** [comment]
`set the start_time
 while True:
        if start_time - current_time >= 15 minutes:
                export the 15 minutes data to the excel file
                start_time = current_time`

**Comment by ian-r-rose** [comment]
Yes, that sounds right to me.

**Comment by vahedi25411** [comment]
Greate. 
Instead of doing that calculation to find the time to export the counters, I wanted to use the following approach:

import time

while True:
____ timer.sleep(15*60)
____ export the counters
